lib/ui/viewer/gallery/gallery_app_bar_widget.dart

```diff
@@ -178,13 +177,45 @@ class _GalleryAppBarWidgetState extends State<GalleryAppBarWidget> {
               await _renameAlbum(context);
             }
             if (value == 2) {
-              await changeCollectionVisibility(
-                context,
-                widget.collection,
-                widget.collection.isArchived()
-                    ? kVisibilityVisible
-                    : kVisibilityArchive,
-              );
+              _isNotHiddenAndIsShared(widget.collection)
+                  ? showDialog(
```

Comment:
Shall we use Choice dialog here?

var choice = await showChoiceDialog(
            context,
            "Hide shared album?",
            'Looks like you're trying to hide a shared album. The album will be hidden on your device, but it can still be seen by the people you have shared it with.',
            firstAction: 'Cancel',
            secondAction: 'Hide',
          );
          if(choice != DialogUserChoice.secondChoice) { 
          return;
           }